### ISSUE TITLE
Include local "modules" to react-app-rewired

### ISSUE THREAD

**Issue by surfer19** [timarney/react-app-rewired#432]
hey,
I have 3 projects(separate repositories):
`project-main`: using react-app-rewired (typescript version) 
`project-module1`: different react repo (also ts)
`project-module2`: different react repo (also ts)
`project-module1` and `project-module2` are linked with yarn to `project-main` and creates bundle inside of main project with ts compiler.

Current solution:
<img width="974" alt="Screenshot 2019-12-19 at 17 29 23" src="https://user-images.githubusercontent.com/10733034/71190781-807ea680-2285-11ea-9c82-f47aaebcae21.png">

What I want to achieve:
<img width="966" alt="Screenshot 2019-12-19 at 17 29 14" src="https://user-images.githubusercontent.com/10733034/71190823-8d02ff00-2285-11ea-9c1d-d579fdfed43f.png">

Basically I need to get rid of part where modules compiles themselves to main project and instead that connect them somehow to main repository compiler in real time. So whenever there will be local code change in folders `project-module1` or `project-module2`, change will be automatically reflected on localhost.
There is also idea connect `project-main` server to modules servers but I think it would be extremely complicated.
What do you think please?

**Comment by nickmccurdy** [comment]
Have you tried npm/yarn link?

**Comment by cbourdage** [comment]
@surfer19 Have you figured out a solution for the above? I'm currently looking at trying to find a way to achieve this same setup. So far I've tried:
* removing the `ModuleScopePlugin` from `config-override.js` and using relative imports from `project-main` to `project-module1`
* adding `resolve.alias` mappings for `project-module1` to `project-main` `config-override.js` file

Any thoughts or suggestions around this would be awesome! I've even given some thought to trying out babel macros to preprocess `project-module*` and handle transpiling there but I fear that would lose out on HMR potentially and for my use-cases here I'd really love to keep that.

**Comment by surfer19** [comment]
Well, you can go for yarn link solution but I don’t recommend it while it’s not easy maintainable dependency approach. 
Try to search for micro-front end approach, if it fits for you. You could wrap every module to web component and include it in runtime. Doesn’t matter if module is written in angular, react or whatever. Even more Easy check framework for that. https://single-spa.js.org/

**Comment by surfer19** [comment]
Idea of this approach is to create completely separate modules that doesn’t share data between themself. Passing data  down main-project->module is allowed.
